client/src/components/ElementEditor/ElementEditor.js

```diff
@@ -22,7 +35,7 @@ class ElementEditor extends PureComponent {
 ElementEditor.propTypes = {
   elementTypes: PropTypes.arrayOf(elementTypeType).isRequired,
   pageId: PropTypes.number.isRequired,
-  baseAddHref: PropTypes.string.isRequired,
+  elementalAreaId: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
```

Comment:
It'd be nice to have some better type safety on this prop.

Reply:
I'd be nice to have return types be consistent, yes. I may be confusing my JS and PHP here though, is it _always_ an Int in React? I've seen errors in both systems.

Reply:
We can cast it before we inject the prop if you want to just make it `number`. IDs in SilverStripe can be strings or ints in different environments =D

Actually if this comes from a GraphQL query then we may not be able to do this gracefully. This is what the GraphQL schema says about the ID type:

>The ID type appears in a JSON response as a String; however, it is not intended to be human-readable. When expected as an input type, any string (such as "4") or integer (such as 4) input value will be accepted as an ID.

Reply:
I'm not certain of the outcome on this comment :/

Reply:
If you can control the input (e.g. from entwine hydration) then cast it to a number and make this strict, if you can't control it then leave it like this =)